Dockerfile

```diff
@@ -1,6 +1,7 @@
+# syntax=docker.io/docker/dockerfile:1.7-labs
 ARG GOLANG_VERSION=1.22.5
 ARG CMAKE_VERSION=3.22.1
-ARG CUDA_VERSION_11=11.3.1
+ARG CUDA_VERSION_11=11.7.1
```

Comment:
11.7.1 was chosen for consistency with upstream llama.cpp. 11.8.0 is actually latest available in the 11 major cuda version.

looking at https://docs.nvidia.com/cuda/archive/11.8.0/cuda-toolkit-release-notes/index.html, not seeing anything hugely compelling outside of rocky 9  support:

> 11.8
> This release introduces support for both the Hopper and Ada Lovelace GPU families.
> Added support for Rocky Linux 9.
> Added support for Kylin OS.
> Package upgradable CUDA is now available starting CUDA 11.8 for Jetson devices. Refer to https://docs.nvidia.com/cuda/cuda-for-tegra-appnote/index.html#upgradable-package-for-jetson for details on how to upgrade to the latest CUDA version on Jetson and the supported JetPack versions.****